src/lib/utils/hooks/SynapseAPI/oauth/useOAuthClient.ts

```diff
@@ -19,3 +30,67 @@ export function useGetOAuthClientInfinite(
     },
   )
 }
+
+export function useDeleteOAuthClient(
+  options?: UseMutationOptions<void, SynapseClientError, string>,
+) {
+  const queryClient = useQueryClient()
+  const { accessToken } = useSynapseContext()
+
+  return useMutation<void, SynapseClientError, string>(
+    (clientId: string) =>
+      SynapseClient.deleteOAuthClient(clientId, accessToken!),
+    {
+      ...options,
+      onSuccess: async (updatedClient, clientId, ctx) => {
+        await queryClient.invalidateQueries(oAuthQueryKeys.all(accessToken!))
+        queryClient.setQueryData(
+          oAuthQueryKeys.edit(accessToken!, clientId),
+          updatedClient,
+        )
+        if (options?.onSuccess) {
+          await options.onSuccess(updatedClient, clientId, ctx)
+        }
+      },
+    },
+  )
+}
+
+export function useMutateOAuthClient(
```

Comment:
I think it might be cleaner to split this into two hooks, `useCreateOAuthClient` and `useEditOAuthClient`. I think this is the first place where we're using mutations for a "create" action, so I'm definitely open to hearing why this might be better, if you think so 🙂

Reply:
This was something I saw when I was working on `mtb` which used a hook for creating and updating cases along with others. I thought this would look "cleaner" since it would be less redundant when importing and calling it for `CreateOAuthClient.tsx`.